internal/sources/mysql/mysql_test.go

```diff
@@ -145,9 +230,38 @@ func TestFailParseFromYaml(t *testing.T) {
 				t.Fatalf("expect parsing to fail")
 			}
 			errStr := err.Error()
-			if errStr != tc.err {
-				t.Fatalf("unexpected error: got %q, want %q", errStr, tc.err)
+			if tc.contains {
+				if !strings.Contains(errStr, tc.err) {
+					t.Fatalf("unexpected error: got %q, want substring %q", errStr, tc.err)
+				}
+			} else {
+				if errStr != tc.err {
+					t.Fatalf("unexpected error: got %q, want %q", errStr, tc.err)
+				}
 			}
 		})
 	}
 }
+
+// TestInitializeInvalidQueryTimeout verifies that an invalid QueryTimeout string is rejected during initialization without attempting a DB connection.
+func TestInitializeInvalidQueryTimeout(t *testing.T) {
```

Comment:
```suggestion
// TestFailInitialization test error during initialization without attempting a DB connection.
func TestFailInitialization(t *testing.T) {
```

Let's provide a more generic name so this could be extended in the future.